### DISCUSSION TITLE
Add contact assignment and create contact with netbox rest api

### DISCUSSION THREAD

**Discussion by sofialouisyax** [netbox-community/netbox#13537] [Q&A] [answered]
Hi!

I'm going through the REST API documentation for Netbox to find how I can
* create contact
* add contact assignment to devices I create via the rest api

I cannot fins anything in the rest api that relates to contacts though. Does this not exist?

**Comment by candlerb** [comment, marked as answer]
What Netbox version? Are you looking at the online, auto-generated REST API docs? These are linked from the bottom of each page:
![image](https://github.com/netbox-community/netbox/assets/44789/a551ec1d-92ad-48b8-bed4-8d6449c4d00a)

Here you will find:
![image](https://github.com/netbox-community/netbox/assets/44789/cd63f574-72fe-4c87-a454-f13464a2a972)
![image](https://github.com/netbox-community/netbox/assets/44789/4513680d-e8a0-4a1f-9ce2-e0e61eda139e)

It would be good if Contact Assignments could be documented under the [Tenancy models](https://docs.netbox.dev/en/stable/models/tenancy/contact/) though.

**Reply by sofialouisyax** [reply]
I was looking here:

https://netbox.grid.uchicago.edu/api/docs/

that don't have it.

![image](https://github.com/netbox-community/netbox/assets/127190270/af2e301d-c057-46ad-bee1-896cee92445f)

But maybe that is now the correct documentation.

I got a tip to go to the documentation from the Netbox UI, which shows the same as your answer

**Reply by sofialouisyax** [reply]
I could not find any link to api documentation from here:

https://docs.netbox.dev/en/stable/integrations/rest-api/

but I found that swagger-page from googling around.

**Reply by candlerb** [reply]
uchicago is running an ancient version of Netbox (v2.8.6, which is more than 3 years old)

For a current version of the API docs, just login to your own Netbox instance, and click on the icon that I indicated at the bottom of the home page.

Or you can use the demo instance: https://demo.netbox.dev/api/schema/swagger-ui/ (login 'admin' and 'admin' if required)

**Comment by candlerb** [comment]
> I could not find any link to api documentation from here:
> 
> https://docs.netbox.dev/en/stable/integrations/rest-api/

Near the top of the page, at https://docs.netbox.dev/en/stable/integrations/rest-api/#interactive-documentation

> #### Interactive Documentation
> 
> Comprehensive, interactive documentation of all REST API endpoints is available on a running NetBox instance at `/api/schema/swagger-ui/`. This interface provides a convenient sandbox for researching and experimenting with specific endpoints and request types. The API itself can also be explored using a web browser by navigating to its root at `/api/`."
